src/util/queries.js

```diff
@@ -0,0 +1,161 @@
+import { gql } from '@apollo/client';
+
+export const QUERY_CHARACTERS = gql`
+query Characters {
+  characters {
+    id
+    thumb
+    art
+    name
+    title
+    rarity
+    type
+    cost
+    ls_description
+    ls_description_eza
+    sa_type
+    sa_name
+    sa_description
+    sa_description_eza
+    ultra_sa_name
+    ultra_sa_description
+    ultra_sa_description_eza
+    ps_name
+    ps_description
+    ps_description_eza
+    sa_type_active
+    active_skill_name
+    active_skill
+    active_skill_condition
+    active_skill_condition_eza
+    transform_type
+    transform_condition
+    transform_condition_eza
+    link_skill
+    category
+    jp_date
+    glb_date
+    jp_date_eza
+    glb_date_eza
+  }
+}
+`;
+
+export const QUERY_ONECHARACTER = gql`
+query Character($dokkanId: Int) {
+  character(dokkanId: $dokkanId) {
+    id
+    thumb
+    art
+    name
+    title
+    rarity
+    type
+    cost
+    ls_description
+    ls_description_eza
+    sa_type
+    sa_name
+    sa_description
+    sa_description_eza
+    ultra_sa_name
+    ultra_sa_description
+    ultra_sa_description_eza
+    ps_name
+    ps_description
+    ps_description_eza
+    sa_type_active
+    active_skill_name
+    active_skill
+    active_skill_condition
+    active_skill_condition_eza
+    transform_type
+    transform_condition
+    transform_condition_eza
+    link_skill
+    category
+    jp_date
+    glb_date
+    jp_date_eza
+    glb_date_eza
+  }
+}
+`;
+
+export const QUERY_LINKS = gql`
+query CharactersLink($linkskill: String) {
+    charactersLink(linkskill: $linkskill) {
+      id
+      link_skill
+    }
+  }
+`;
+
+export const QUERY_6LINKS = gql`
+query Characters6Link($link1: String, $link2: String, $link3: String, $link4: String, $link5: String, $link6: String) {
+  characters6Link(link1: $link1, link2: $link2, link3: $link3, link4: $link4, link5: $link5, link6: $link6) {
+    id
+    link_skill
+  }
+}`;
+
+export const QUERY_7LINKS = gql`
+query Characters7Link($link1: String, $link2: String, $link3: String, $link4: String, $link5: String, $link6: String, $link7: String) {
+  characters7Link(link1: $link1, link2: $link2, link3: $link3, link4: $link4, link5: $link5, link6: $link6, link7: $link7) {
+    id
+    link_skill
+    thumb
+    art
+    type
+    rarity
+  }
+}`;
+
+export const GET_USERDATA = gql`
+query Me($username: String!) {
+  me(username: $username) {
+    username
+    email
+    savedCharacters
+  }
+}`;
+
+export const GET_USERCHARACTERSBYID = gql`
```

Comment:
I think I left this comment elsewhere, but if the intent is to query for all user character objects, I'd recommend adding logic to the serverside resolver to query for the user's character ids, then query the character objects all in one query (IIRC you're using mongo, so you could do this with multiple queries or an [aggregation pipeline](https://www.mongodb.com/docs/manual/aggregation/). See [match](https://www.mongodb.com/docs/manual/reference/operator/aggregation/match/#-match--aggregation-))

As an aside, this exact problem is a weakness for mongo, so it could be worth thinking about redesigning the datastore. For this case it's not a big deal to make 2 queries, but here's some alternative ways of solving the problem:
* rather than store character ids for each user, store an array of `userSaved` on the character document itself. Then to query for all user characters, query the characters collection for `{ userSaved: currentUsername }`. Saving/removing a character for a user is just adding or removing the current username from that array. Downside is that if you get a lot of users, this becomes a performance problem even if you put an index on the array
* if the character data is effectively immutable, rather than storing a reference id per user, you could just duplicate the character object within each user's characters so that querying for a user's characters will just get all the data. Downside is that if the character data ever changes, you now have inconsistent data and it will be hard to keep in sync without a lot more work
* Switch to a relational datastore. Downside is a lot of up front effort, plus you lose all the benefits of the document style DB that you're currently able to take advantage of